CMake/itkCompilerChecks.cmake

```diff
@@ -1,31 +1,33 @@
-# Minimum compiler version check: GCC >= 4.8
+# Minimum compiler version check: GCC
 if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND
-    CMAKE_CXX_COMPILER_VERSION VERSION_LESS 4.8)
-  message(FATAL_ERROR "GCC 4.8 or later is required.")
+    CMAKE_CXX_COMPILER_VERSION VERSION_LESS 5.1)
+  message(FATAL_ERROR "GCC 5.1 or later is required.")
 endif ()
 
-# Minimum compiler version check: LLVM Clang >= 3.3
+# Minimum compiler version check: LLVM Clang
 if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND
-    CMAKE_CXX_COMPILER_VERSION VERSION_LESS 3.3)
-  message(FATAL_ERROR "LLVM Clang 3.3 or later is required.")
+    CMAKE_CXX_COMPILER_VERSION VERSION_LESS 3.4)
+  message(FATAL_ERROR "LLVM Clang 3.4 or later is required.")
 endif ()
 
-# Minimum compiler version check: Apple Clang >= 5.0 (Xcode 5.0)
+# Minimum compiler version check: Apple Clang >= 5.1 (Xcode 5.1)
 if (CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang" AND
-    CMAKE_CXX_COMPILER_VERSION VERSION_LESS 5.0)
-  message(FATAL_ERROR "Apple Clang 5.0 or later is required.")
+    CMAKE_CXX_COMPILER_VERSION VERSION_LESS 5.1)
+  message(FATAL_ERROR "Apple Clang 5.1 or later is required.")
 endif ()
 
 # Minimum compiler version check: Microsoft C/C++ >= 19.0 (aka VS 2015 aka VS 14.0)
+# Note that VS 2015 has implemented most C++14 features, although it still does not
```

Comment:
@blowekamp do we agree we can drop v140 toolkit and move to v141 (VS2017)?

Reply:
We can drop the v140 toolkit. Even VS2015 has been updated to use v141 by default now.

Reply:
Awesome! @N-Dekker please bump to VS2017!

Reply:
> please bump to VS2017!

@dzenanz Amended and force-pushed, please check: https://github.com/InsightSoftwareConsortium/ITK/compare/a35dcbb8d8469663fd58ef23765380c056f13eb2..40a795f019427f94a06e1b295b0ab66502377dc2